CenterForOpenScience/osf.io
# [feature] Edit wiki pages via APIv2 [PLAT-543]

Initial PR/review comments: https://github.com/pattisdr/osf.io/pull/10

#### Purpose
* Adds writeable wiki endpoints to APIv2.
  * Create a wiki page. 
  * Rename a wiki page.
  * Delete a wiki page.
  * Edit a wiki page (create a new version)

#### Changes
- Create a new wiki (`content` can optionally be specified here)
  - POST `/nodes/<node_id>/wikis/` (NodeWikiList)
- Edit existing wiki content
  - POST `/wikis/<id>/versions/` (WikiVersionList) 
- Delete a wiki
  - DELETE `/wikis/<id>/` (WikiDetail)
- Rename a wiki
  - PATCH `/wikis/<id>/` (WikiDetail)

#### QA Notes
- At the moment, manual testing isn't possible since the front end doesn't use this functionality. This PR contains basic tests to ensure that permissions are handled and each endpoint functions as expected. Runscope tests could be added in the future. 

#### Documentation
- https://github.com/CenterForOpenScience/developer.osf.io/pull/8

#### Side Effects
- None expected.

#### Ticket
- [PLAT-543](https://openscience.atlassian.net/browse/PLAT-543)
